better_bookmarks/js/content_script.js

```diff
@@ -0,0 +1,28 @@
+var port = chrome.runtime.connect();
+
+window.addEventListener("message", (event) => {
+    if (event.source != window)
+        return;
+
+    if (event.data.type && (event.data.type == "FROM_PAGE")) {
+        console.log("Content script received: " + event.data.text);
+        port.postMessage(event.data.text);
+    }
+}, false);
+
+const title = document.getElementsByTagName("title")[0];
+const metas = document.getElementsByTagName("meta");
+const url = document.URL;
+for (let i = 0; i < metas.length; i++) {
+    const meta = metas[i];
+    if (meta.name.toLowerCase() == "description"){
+        var description = meta.content;
+    }
+}
+if(description === undefined){
+    chrome.storage.sync.set({title: title.innerHTML, desc: "", url: url}, function () {});
```

Comment:
With the way that JavaScript objects work, we can leave the ```desc: ""``` part out, and instead write: ```{title: title.innerText, url: url}```.  It will be Kyle's job to ensure that the description value is present when he makes an attempt to access this data.

The ```innerText``` property treats the underlying value as text rather than HTML content, so in this scenario, I think ```innerText``` is more appropriate since it conveys more of what we intend the title value to be.

Reply:
tl:dr If i dont set description to "", then it will get the last description that was set.

Let's say I go to a tab, tab1, with a description. Then it will store `title: tab1_title, desc: tab1_description, url: tab1_url`. If I then go to a second tab, tab2, that doesn't have a description. When it goes to store the info of the new tab, it overwrite the title key and the url key but not the description key. So when I go `storage.sync.get("desc" ...` it will pull the description that was last stored as in `tab1_description`.

Reply:
Actually, I just noticed that this whole thing can be reduced to one line:
    Instead of checking if ```description``` is ```undefined```, just pass it along without checking, and leave it up to Kyle to check in his function.

Also, switch to using ```innerText``` instead of ```innerHTML``` since that's what we're supposed to be passing to the ML functions.  There may be a case where HTML is in the text, and it could show up using ```innerHTML```, which we don't want.  See this link for details: https://www.geeksforgeeks.org/difference-between-innertext-and-innerhtml/.